download/oci_download_test.go

```diff
@@ -0,0 +1,105 @@
+//go:build slow
+// +build slow
+
+package download
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestOCIStartStop(t *testing.T) {
+	ctx := context.Background()
+	fixture := newTestFixture(t)
+	fixture.server.expAuth = "" // test on public registry
+	fixture.server.expEtag = "sha256:c5834dbce332cabe6ae68a364de171a50bf5b08024c27d7c08cc72878b4df7ff"
+
+	updates := make(chan *Update)
+
+	config := Config{}
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+	// TODO: remove restConfig and actual rest client - for now it can be used to make an actual call to a public repository by replacing the fixtures.client with this
+	// was used as a dev test harness
+	/*
+			restConfig := []byte(fmt.Sprintf(`{
+						"name":"opcr",
+						"url": "%s",
+						"credentials": {
+						}
+					}`, "https://opcr.io"))
+
+			tc, err := rest.New(restConfig, map[string]*keys.Config{})
+
+			if err != nil {
+				t.Fatal(err)
+			}
+
+		d := NewOCI(config, tc, "opcr.io/dani/ppf-rbac:0.0.1", "/tmp/opa/").WithCallback(func(_ context.Context, u Update) {
+			updates <- &u
+		})
+	*/
+	d := NewOCI(config, fixture.client, "ghcr.io/org/repo:latest", "/tmp/opa/").WithCallback(func(_ context.Context, u Update) {
```

Comment:
So this makes a call to the actual external registry? I'd rather avoid that in our tests... Can we leverage any test fixtures provided by the introduced dependencies, maybe?

Update: looks like that's what we're setting up in `newTestFixture`, aren't we? Just want to be sure.